airflow/bin/cli.py

```diff
@@ -368,24 +367,22 @@ def run(args, dag=None):
         # tries to write to a log file created by the other user.
         log_base = os.path.expanduser(conf.get('core', 'BASE_LOG_FOLDER'))
         directory = log_base + "/{args.dag_id}/{args.task_id}".format(args=args)
+        # Create the log file and give it group writable permissions
+        # TODO(aoen): Make log dirs and logs globally readable for now since the SubDag
+        # operator is not compatible with impersonation (e.g. if a Celery executor is used
+        # for a SubDag operator and the SubDag operator has a different owner than the
+        # parent DAG)
         if not os.path.exists(directory):
             # Create the directory as group writable using custom mkdirs
             # as os.makedirs doesn't set mode properly.
             mode = 0x775
-            mkdirs(directory, 0o775)
+            mkdirs(directory, 0o777)
         iso = args.execution_date.isoformat()
         filename = "{directory}/{iso}".format(**locals())
 
-        # TODO:: remove this after Impersonation migration is stable
-        # Ensure that the log file is writable by airflow_sudoable
-        subprocess.call(
-            ['sudo', 'chown', "{}".format(os.getuid()), filename]
-        )
-
-        # Create the log file and give it group writable permissions
         if not os.path.exists(filename):
             open(filename, "a").close()
-            mode = 0o664
+            mode = 0o666
```

Comment:
A log file does not have to be world executable. 666 is really a terrible setting.